test/test_helper.rb

```diff
@@ -30,3 +30,15 @@ def with_preview_route(new_value)
   Rails.application.config.view_component.preview_route = old_value
   app.reloader.reload!
 end
+
+def without_render_monkey_patch
+  begin
+    old_value = Rails.application.config.view_component.render_monkey_patch_enabled
+    Rails.application.config.view_component.render_monkey_patch_enabled = false
```

Comment:
Setting this option here to `false` does not work since the initializers of `ViewComponent::Engine` only run once when the rails app gets booted.  Still not sure how to solve this. Recommendations are welcome. 

Found this thread on StackOverflow giving different options of how we could solve this: 
https://stackoverflow.com/questions/37815826/how-to-tell-rspec-to-restart-rails-before-an-example-is-run

Reply:
@johannesengl I wouldn't be opposed to setting up a second Rails app in the test suite for this compatibility mode.

Reply:
@joelhawksley 
I played around with the test setup for a while and tried two approaches:

**Second test dummy app**
In my option creating a new dummy application each time we want to test the behavior of config options changing the initializing process of the engine is not scalable.

**Swapping the `RAILS_ENV` in the rake test task**
I like the approach of swapping the RAILS_ENV more since we don't have to maintain two full dummy applications. 

What is your opinion regarding this? Any idea how we could solve this in a more elegant way? 

Done in d6f9fa91f0dda9930df119306028e2c0092b0a3a

Reply:
@johannesengl I think a second dummy app is the way to go here, given the downsides you've identified with the `RAILS_ENV` approach. I don't see us having many other cases where this kind of configuration change will need to be tested.